dapp/src/components/dashboard/components/RewardUserForm/index.jsx

```diff
@@ -1,89 +1,64 @@
-import React, { Fragment, Component } from 'react'
+import React, { Component } from 'react'
 import classNames from 'classnames'
-import TextField from '@material-ui/core/TextField'
 import { Formik } from 'formik'
-import { object, number, boolean } from 'yup'
+import isEqual from 'lodash/isEqual'
+import bonusesShape from 'utils/validation/shapes/bonuses'
+import Options from './Option'
 
 class RewardUserForm extends Component {
   constructor (props) {
     super(props)
 
     this.initialValues = {
-      amount: '',
       ...this.props.initialValues
     }
 
-    this.validationSchema = object().noUnknown(false).shape({
-      amount: number().min(0),
-      activated: boolean()
-    })
-  }
-
-  componentDidUpdate (prevProps) {
-    if (prevProps.initialValues !== this.props.initialValues) {
-      this.initialValues = {
-        amount: '',
-        ...this.props.initialValues
-      }
-    }
+    this.validationSchema = bonusesShape
   }
 
   onSubmit = (values, formikBag) => {
-    const { amount } = values
-    const { setBonus } = this.props
-    setBonus(amount)
-    formikBag.resetForm({ amount })
+    const { initialValues } = this.props
+    if (!isEqual(initialValues.joinBonus, values.joinBonus)) {
+      this.props.setJoinBonus(values.joinBonus.amount, values.joinBonus.isActive)
+      formikBag.resetForm(values)
```

Comment:
not sure why do you need to reserForm on each change. Something technical?

Reply:
Yeah you're right.
I fixed it.